deploy/eck-stack/charts/eck-beats/templates/beats.yaml

```diff
@@ -16,4 +16,4 @@ spec:
   {{- if and (not (hasKey .Values.spec "daemonSet")) (not (hasKey .Values.spec "deployment")) }}
   {{ fail "At least one of daemonSet or deployment is required for a functional Beat" }}
   {{- end }}
-  {{- toYaml .Values.spec | nindent 2 }}
+  {{- unset .Values.spec "type" | toYaml | nindent 2 }}
```

Comment:
Our issue comes from the fact that we want to validate the `type` is set and we use `required` for that, which checks and renders the value at the same time.

No strong opinion but it seems a bit twisted to keep `required` and therefore remove the field afterwards.

```yaml
spec:
  type: {{ required "A Beat type is required" .Values.spec.type }}
  {{- unset .Values.spec "type" | toYaml | nindent 2 }}
```

I would do the check with an `if/fail` instead of `required` and keep the rendering the entire spec.

```yaml
spec:
  {{- if not .Values.spec.type }}{{ fail "A Beat type is required" }}{{- end }}
  {{- toYaml .Values.spec | nindent 2 }}
```

An alternative would be to move the `spec.type` value to the root level like we did for `version`.